### ISSUE TITLE
Cannot connect Tor Browser to Keepass-XC

### ISSUE THREAD

**Issue by jpcuzz** [keepassxreboot/keepassxc#6465]
KeePassXC - Version 2.6.4
Revision: 34a78f0
Distribution: AppImage

Qt 5.15.2
Debugging mode is disabled.

Operating system: Debian GNU/Linux 10 (buster)
CPU architecture: x86_64
Kernel: linux 4.19.0-16-amd64

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey
- Secret Service Integration

Cryptographic libraries:
- libgcrypt 1.8.1

Here are the errors from the tor-browser plugin:

Cannot connect to KeePassXC. Check that browser integration is enabled in KeePassXC settings. 

KeePassXC-Browser has encountered an error:

Key exchange was not successful. 

I've been unable to find a good way to troubleshoot this error.

Any help with steps I should take?
I'm running Whonix on a Linux host.
Standard Tor-browser on Whonix workstation.
Tor Browser 10.0.16 (based on Mozilla Firefox 78.20.0esr) (64-bit)

Thanks for any help!!

**Comment by droidmonkey** [comment]
Use the custom browser option in the advanced browser integration settings. This has been asked several times before so you can search our issues list for more information. Basically tor places the native messaging folder in numerous different locations depending on the installer used.

**Comment by Humpkes** [comment]
In case it helps somebody ... I did find several posts about this issue but it didn't seem to work out for me. What I did to make it work:

 `cd /usr/local/share/tor-browser/Browser/TorBrowser/Data/Browser/`
 `mkdir .mozilla`
 `cd .mozilla`
 `mkdir native-messaging-hosts`
 `cd native-messaging-hosts`
 `cp ~/.mozilla/native-messaging-hosts/org.keepassxc.keepassxc_browser.json .`

It can be specific to my system. I run Linux Mint. After copying that json-file the connect button popped up and everything was okay.  I did enable Tor Browser in the Browser Integration tab of the KeepassXC-settings.
